From 2f41c8176336ef2cbec17752ea06c2471a9ec7dc Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Sat, 07 Nov 2020 23:54:01 +0200
Subject: [PATCH] [SW-20806] i2c: use sized types instead of native

In preparation to I2C refactoring and addition of write functionality,
use sized type for more precise memory usage and uniformity.

Change-Id: Icd32622ae330da6aa9a61ccf1861e187554eb0f6
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/common/module_i2c_slave.c b/common/module_i2c_slave.c
index 1592157..9ad75ff 100644
--- a/common/module_i2c_slave.c
+++ b/common/module_i2c_slave.c
@@ -133,7 +133,7 @@
 enum cmd_check_res module_i2c_s_check(void)
 {
 	enum cmd_check_res res = NOT_FOUND;
-	unsigned int stat;
+	u32 stat;
 
 	if (hl_is_debug_mode(HL_DEBUG_HW_I2C))
 		return res;
@@ -157,10 +157,10 @@
 	preboot_puts("I2C SLAVE: disabled\n");
 }
 
-char i2cslave_pcie_vendor_id_cmd(int command_idx)
+u8 i2cslave_pcie_vendor_id_cmd(u8 command_idx)
 {
-	char ret;
-	unsigned int data = regmap_readl(mmPCIE_DBI_DEVICE_ID_VENDOR_ID_REG);
+	u8 ret;
+	u32 data = regmap_readl(mmPCIE_DBI_DEVICE_ID_VENDOR_ID_REG);
 
 	switch (command_idx) {
 	case HL_PCIE_VENDOR_ID_VAL_LOW:
@@ -181,8 +181,8 @@
 
 void module_i2c_s_run(void)
 {
-	unsigned int stat;
-	int ret;
+	u32 stat;
+	u8 ret;
 
 	stat = regmap_readl(mmPSOC_I2C_S_IC_INTR_STAT);
 
@@ -209,7 +209,7 @@
 
 void module_i2c_s_teardown(void)
 {
-	unsigned int val;
+	u32 val;
 
 	/* Write 0 to IC_ENABLE to disable DW_apb_i2c */
 	regmap_writel(0, mmPSOC_I2C_S_IC_ENABLE);
diff --git a/common/module_i2c_slave.h b/common/module_i2c_slave.h
index a5a66c3..c1957c9 100644
--- a/common/module_i2c_slave.h
+++ b/common/module_i2c_slave.h
@@ -9,11 +9,11 @@
 #define _MODULE_I2C_SLAVE_H_
 
 struct module_i2c_data {
-	unsigned int cmd;
-	unsigned int read_bytes_count;
+	u8 cmd;
+	u32 read_bytes_count;
 };
 
-char i2cslave_pcie_vendor_id_cmd(int command_idx);
-int i2cslave_handle_read_req(struct module_i2c_data *data);
+u8 i2cslave_pcie_vendor_id_cmd(u8 command_idx);
+u8 i2cslave_handle_read_req(struct module_i2c_data *data);
 
 #endif /* _MODULE_I2C_SLAVE_H_ */
diff --git a/preboot/module_i2c_slave.c b/preboot/module_i2c_slave.c
index 627b37c..1b24ec0 100644
--- a/preboot/module_i2c_slave.c
+++ b/preboot/module_i2c_slave.c
@@ -11,9 +11,10 @@
 #include "thermal.h"
 #include "module_i2c_slave.h"
 
-int i2cslave_handle_read_req(struct module_i2c_data *data)
+u8 i2cslave_handle_read_req(struct module_i2c_data *data)
 {
-	int max_tmp, ret;
+	int max_tmp;
+	u8 ret;
 
 	switch (data->cmd) {
 	case HL_ASIC_TEMP_CMD_CODE:
diff --git a/recovery/module_i2c_slave.c b/recovery/module_i2c_slave.c
index 0151dc3..3cb1470 100644
--- a/recovery/module_i2c_slave.c
+++ b/recovery/module_i2c_slave.c
@@ -10,9 +10,9 @@
 
 #include "module_i2c_slave.h"
 
-int i2cslave_handle_read_req(struct module_i2c_data *data)
+u8 i2cslave_handle_read_req(struct module_i2c_data *data)
 {
-	int ret;
+	u8 ret;
 
 	switch (data->cmd) {
 	case HL_FW_UPDATE_CMD_CODE:
